test/Spec/BotPlutusInterface/CoinSelection.hs

```diff
@@ -0,0 +1,229 @@
+{-# OPTIONS_GHC -Wno-unused-binds #-}
+
+module Spec.BotPlutusInterface.CoinSelection (tests) where
+
+import BotPlutusInterface.CoinSelection (selectTxIns, uniqueAssetClasses, valuesToVecs)
+import BotPlutusInterface.Effects (PABEffect)
+import Control.Lens (filtered, foldOf, folded, to, (^..))
+import Data.Default (def)
+import Data.Either (isLeft)
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Data.Maybe (fromJust)
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Vector (Vector)
+import Data.Vector qualified as Vec
+import Ledger qualified
+import Ledger.Ada qualified as Ada
+import Ledger.Address (Address (Address), PaymentPubKeyHash (PaymentPubKeyHash))
+import Ledger.Address qualified as Address
+import Ledger.CardanoWallet qualified as Wallet
+import Ledger.Crypto (PubKeyHash)
+import Ledger.Tx (TxIn (..), TxInType (..), TxOut (..), TxOutRef (..))
+import Ledger.Value (AssetClass, Value, leq)
+import Ledger.Value qualified as Value
+import Plutus.V1.Ledger.Credential (Credential (PubKeyCredential, ScriptCredential))
+import Spec.MockContract (runPABEffectPure)
+import Spec.RandomLedger
+import Test.QuickCheck (Gen, Property, forAll, withMaxSuccess)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (Assertion, assertFailure, testCase, (@?=))
+import Test.Tasty.QuickCheck (testProperty)
+import Prelude
+
+tests :: TestTree
+tests =
+  testGroup
+    "BotPlutusInterface.CoinSelection"
+    [ testProperty "Have All unique assetClasses" assertUniqueAssetClasses
+    , testProperty "columns of vectors represent same assetClass" validValueVectors
+    , testProperty "coin selection results in valid balance of Tx" validateBalancing
```

Comment:
Was it supposed to be "coin selection results **is**  valid balance of Tx"?

Reply:
No, sorry for name the test poorly. It was supposed to mean: "coin selection produces valid balanced Tx".